pkgs/os-specific/darwin/apple-source-releases/default.nix

```
@@ -135,35 +138,47 @@ let
     };
   };
 
-  fetchApple = version: sha256: name: let
+  fetchApple_ = pname: version: sha256: let
     # When cross-compiling, fetchurl depends on libiconv, resulting
     # in an infinite recursion without this. It's not clear why this
     # worked fine when not cross-compiling
-    fetch = if name == "libiconv"
+    fetch = if pname == "libiconv"
       then stdenv.fetchurlBoot
       else fetchurl;
   in fetch {
-    url = "http://www.opensource.apple.com/tarballs/${name}/${name}-${versions.${version}.${name}}.tar.gz";
+    url = "http://www.opensource.apple.com/tarballs/${pname}/${pname}-${version}.tar.gz";
     inherit sha256;
   };
 
-  appleDerivation_ = name: version: sha256: attrs: stdenv.mkDerivation ({
-    inherit version;
-    name = "${name}-${version}";
+  fetchApple = sdkName: sha256: pname: let
+    version = versions.${sdkName}.${pname};
+  in fetchApple_ pname version sha256;
+
+  appleDerivation_ = pname: version: sdkName: sha256: attrs: stdenv.mkDerivation ({
```

We could keep as is, but might as well change to
```suggestion
  appleDerivation' = pname: version: sdkName: sha256: attrs: stdenv.mkDerivation ({
```